onnxruntime/core/providers/cpu/nn/flatten.h

```diff
@@ -21,9 +22,17 @@ class Flatten final : public OpKernel {
     if (X == nullptr) return Status(common::ONNXRUNTIME, common::FAIL, "input count mismatch");
 
     const TensorShape& X_shape = X->Shape();
-    ORT_ENFORCE(gsl::narrow_cast<int64_t>(X_shape.NumDimensions()) >= axis_, "The rank of input tensor must be >= axis");
-
-    Tensor* Y = context->Output(0, TensorShape({X_shape.SizeToDimension(axis_), X_shape.SizeFromDimension(axis_)}));
+    auto axis = axis_;
+    
+    // Valid axis range is [-rank, rank] instead of [-rank, rank-1], add additional check to only handle neg axis case.
+    if (axis < 0)
+    {
```

Comment:
nit: could just call HandleNegativeAxis without this test.

Reply:
No, HandleNegativeAxis requires axis <= rank -1, otherwise it will fail.

ORT_ENFORCE(axis >= -tensor_rank && axis <= tensor_rank - 1, "axis ", axis,
              " is not in valid range [-", tensor_rank, ",", tensor_rank - 1, "]");